contractcourt/htlc_incoming_contest_resolver.go

```diff
@@ -150,7 +150,7 @@ func (h *htlcIncomingContestResolver) Resolve() (ContractResolver, error) {
 	// the sender doesn't sweep and we already have or will learn the
 	// preimage. Otherwise the resolver could potentially stay active
 	// indefinitely and the channel will never close properly.
-	if uint32(currentHeight) >= h.htlcExpiry {
+	if uint32(currentHeight) > h.htlcExpiry+1 {
```

Comment:
Commit message is slightly inaccurate here. For LND, the outgoing broadcast delta is 0, so typically the force close is broadcasted when `currentHeight == h.htlcExpiry` unless there are other HTLCs, user-initiated, etc. Other implementations may broadcast with non-zero deltas though. The earliest the timeout transaction can be confirmed is at height `h.htlcExpiry+1` so the conditional could instead read `if uint32(currentHeight) > h.htlcExpiry` to stop preimage-claiming as soon as the counterparty can time it out. I don't think it makes a big difference here though and we could potentially even bump the `h.htlcExpiry+1` to something larger to account for variable fee environments

Reply:
Cool this commit is now removed. It was added to trigger the race condition used in the itest `testExtraMempoolPreimage`. Now that test will only be added in the fee bumper PR, will also move this commit and its related discussion there.